src/openrct2/peep/Peep.cpp

```diff
@@ -8053,9 +8053,8 @@ rct_peep * peep_generate(sint32 x, sint32 y, sint32 z)
     uint8 trousers_colour = static_cast<uint8>(peep_rand() % Util::CountOf(trouser_colours));
     peep->trousers_colour = trouser_colours[trousers_colour];
 
-    /* It looks like 65 is about 50% energy level, so this initialises
-     * a peep with approx 50%-100% energy (0x3F = 63, 63 + 65 = 128). */
-    uint8 energy        = (peep_rand() & 0x3F) + 65;
+    // Initialise a peep with approx 50%-100% energy
+    uint8 energy        = (peep_rand() & (PEEP_MAX_ENERGY / 2)) + (PEEP_MAX_ENERGY/2);
```

Comment:
This is dangerous. What happens if the max energy does not divide by two nicely. If max energy was 8 this would be 4 which would mask all the bits but 1.

Reply:
PEEP_MAX_ENERGY = 128.

Possibly better to revert this bit.

Reply:
You would have to change and to mod.

Reply:
?

Reply:
it should be `(peep_rand() % (PEEP_MAX_ENERGY / 2)) + PEEP_MAX_ENERGY / 2`

Reply:
Sorry, I mean what @duncanspumpkin said.
Alternatively you can make a new rand function which takes a lower and upper bound which does the same thing.

Reply:
That would yield different results than it does now. To yield the same, it should be 
`(peep_rand() % (PEEP_MAX_ENERGY / 2)) + ((PEEP_MAX_ENERGY / 2) + 1)`

Reply:
It may yield negligibly different results but it simplifies the code by a great deal allowing you to remove the comment above.

E.g.
`uint8 energy = peep_rand(PEEP_MAX_ENERGY / 2, PEEP_MAX_ENERGY);`

Where is `peep_rand(lower, upper)`.